store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -881,10 +881,50 @@ private static long getFileUploadMaxSize(boolean limitByFileUploadMaxSize) {
         return maxSize;
     }
 
+    private static long getFileUploadMaxSize(boolean limitByFileUploadMaxSize, Account acct) {
+        long maxSize = DEFAULT_MAX_SIZE;
+        if (acct.getMailAttachmentMaxSize() >= 0 || acct.getFileUploadMaxSizePerFile() >= 0) {
+            try {
+                if (limitByFileUploadMaxSize) {
+                    long fileUploadSizePerFile = acct.getFileUploadMaxSizePerFile();
+                    long fileUploadSize = Provisioning.getInstance().getLocalServer().getLongAttr(
+                            Provisioning.A_zimbraFileUploadMaxSize, DEFAULT_MAX_SIZE);
+                    if (fileUploadSizePerFile >= fileUploadSize) {
+                        maxSize = fileUploadSize;
+                    } else {
+                        maxSize = fileUploadSizePerFile;
+                    }
+                } else {
+                    long mailAttachmentMaxSize = acct.getMailAttachmentMaxSize();
+                    long mtaMaxMsgSize = Provisioning.getInstance().getConfig().getLongAttr(
+                            Provisioning.A_zimbraMtaMaxMessageSize, DEFAULT_MAX_SIZE);
+                    if (mailAttachmentMaxSize > mtaMaxMsgSize) {
+                        maxSize = mtaMaxMsgSize;
+                    } else {
+                        maxSize = mailAttachmentMaxSize;
+                    }
+                    if (maxSize == 0) {
+                        maxSize = -1;
+                    }
+                }
+            } catch (ServiceException e) {
+                // TODO Auto-generated catch block
```

Comment:
do we want to log and re-throw the exception? i think we are not supposed to swallow exception here.